### ISSUE TITLE
Spotifyd won't start at all (raspbian 10)

### ISSUE THREAD

**Issue by hugovantriet** [Spotifyd/spotifyd#555]
The program is not working for me at all, I changed all the settings in the correct way, and I fixed the firewall, but nothing seems to help. I use a Raspberry Pi Zero W, equiped with raspbian buster lite.
When I try the command `sudo systemctl status spotifyd` it keeps saying this:
 ● spotifyd.service - A spotify playing daemon
   Loaded: loaded (/etc/systemd/system/spotifyd.service; enabled; vendor preset: enabled)
   Active: activating (auto-restart) (Result: signal) since Thu 2020-04-02 12:44:51 BST; 4s ago
     Docs: https://github.com/Spotifyd/spotifyd
  Process: 788 ExecStart=/usr/local/bin/spotifyd --no-daemon (code=killed, signal=ILL)
 Main PID: 788 (code=killed, signal=ILL)

I'm not an expert at all, so do you maybe know what I am doing wrong?
Thanks in advance!

**Comment by SimonPersson** [comment]
Can you try to run `spotifyd --no-daemon` from your shell, instead of through sysremd?

**Comment by m0ar** [comment]
@SimonPersson @hugovantriet 
I see the same error from systemd, but invoking `spotifyd --no-daemon` manually sheds gives more clues:

```
pi@mediacenter:~ $ /usr/bin/spotifyd --no-daemon
Loading config from "/home/pi/.config/spotifyd/spotifyd.conf"
No proxy specified
Using alsa volume controller.
error sending packet Os { code: 99, kind: AddrNotAvailable, message: "Cannot assign requested address" }
Connecting to AP "gew1-accesspoint-b-hfcq.ap.spotify.com:443"
Authenticated as "razoz" !
Using alsa sink
Country: "SE"
Caught panic with message: Failed to initialize DBus connection: D-Bus error: Unable to autolaunch a dbus-daemon without a $DISPLAY for X11 (org.freedesktop.DBus.Error.NotSupported)
```

Since @hugovantriet said he is using raspbian lite, it could be the same error since I don't run a DE/WM, just Kodi.

**Comment by jgabriel98** [comment]
> @SimonPersson @hugovantriet
> I see the same error from systemd, but invoking `spotifyd --no-daemon` manually gives more clues:
> 
> ```
> pi@mediacenter:~ $ /usr/bin/spotifyd --no-daemon
> Loading config from "/home/pi/.config/spotifyd/spotifyd.conf"
> No proxy specified
> Using alsa volume controller.
> error sending packet Os { code: 99, kind: AddrNotAvailable, message: "Cannot assign requested address" }
> Connecting to AP "gew1-accesspoint-b-hfcq.ap.spotify.com:443"
> Authenticated as "razoz" !
> Using alsa sink
> Country: "SE"
> Caught panic with message: Failed to initialize DBus connection: D-Bus error: Unable to autolaunch a dbus-daemon without a $DISPLAY for X11 (org.freedesktop.DBus.Error.NotSupported)
> ```
> 
> Since @hugovantriet said he is using raspbian lite, it could be the same error since I don't run a DE/WM, just Kodi.

Any updates on this?

I have the exactly same error, when trying to run o raspbian with desktop (not lite) but as a system service, instead of user service.

if i run `spotifyd --no-daemon` the error wont happen

**Comment by mainmachine** [comment]
I have this same problem on a Pi4, but _**not**_ on a Pi Zero W, both running Raspbian 10 Buster (Lite, so no desktop). I've set up each one almost identically, the only difference at the moment is the Zero W has an i2s DAC hat, while I'm trying to use the headphone out on the Pi4.

Searched for dbus stuff that might be installed with ```dpkg -l  | grep dbus``` and there are only 2 packages, same on both so I don't see that being the cause...

So the only differences I am aware of are the CPU and the DAC versus no DAC.

Anybody want me to test anything on these to see if we can identify the root cause...?

**Comment by mainmachine** [comment]
OK, the cause is that the "full" binary has ```dbus_mpris``` enabled by default. This [issue](https://github.com/Spotifyd/spotifyd/issues/625) has an explanation and a workaround:

```/usr/bin/dbus-run-session /usr/bin/spotifyd --no-daemon```

TL;DR: ```dbus_mpris``` is enabled by default on the "full" binary. Workaround is to modify the systemd service (or script, whatever you're using) as above, recompile omitting the ```dbus_mpris``` or just grab the "slim" binary.

I think this issue can be closed, OP. :)

**Comment by SirWindfield** [comment]
Yep, headless does have dbus, but only system sessions and we use user sessions iirc. There was a feature request to add a cli flag to toggle between them or jsut set the mode but that hasn't been done eyet./
